File: SentrySMP.Domain/Entities/Rank.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SentrySMP.Domain.Entities;

public class Rank
{
    public int Id { get; set; }

    [Required]
    [MaxLength(100)]
    public string Name { get; set; } = string.Empty;

    [MaxLength(500)]
    public string Description { get; set; } = string.Empty;

    public double Price { get; set; }

    public int Sale { get; set; }

    [MaxLength(255)]
    public string? Image { get; set; }
}

File: sentrysmp-laravel/app/Models/Rank.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Rank extends Model
{
    protected $table = 'ranks';
    protected $primaryKey = 'Id';
    public $timestamps = false;

    protected $fillable = ['Name', 'Description', 'Price', 'Sale', 'Image', 'GlobalMaxOrder'];

    public function getEffectivePriceAttribute(): float
    {
        return $this->Sale > 0 ? round($this->Price * (1 - $this->Sale / 100), 2) : $this->Price;
    }
}
